unrouted_handler.go

```diff
@@ -437,12 +450,27 @@ func (handler *UnroutedHandler) PatchFile(w http.ResponseWriter, r *http.Request
 	}
 
 	// Do not proxy the call to the data store if the upload is already completed
-	if info.Offset == info.Size {
+	if !info.SizeIsDeferred && info.Offset == info.Size {
 		w.Header().Set("Upload-Offset", strconv.FormatInt(offset, 10))
 		handler.sendResp(w, r, http.StatusNoContent)
 		return
 	}
 
+	if handler.composer.UsesLengthDeferrer && info.SizeIsDeferred && r.Header.Get("Upload-Length") != "" {
```

Comment:
Although this check is perfectly fine, I would love to see an error if someone tries to declare the length for an upload for a) an upload where length deferring is not enabled and b) an upload where the length has already been declared.